core-common/src/main/java/org/mapstruct/factory/Mappers.java

```diff
@@ -93,11 +97,13 @@ private Mappers() {
         throw new ClassNotFoundException("Cannot find implementation for " + mapperType.getName() );
     }
 
-    private static <T> T doGetMapper(Class<T> clazz, ClassLoader classLoader) {
+    private static <T> T doGetMapper(Class<T> clazz, ClassLoader classLoader) throws NoSuchMethodException {
         try {
             @SuppressWarnings("unchecked")
-            T mapper = (T) classLoader.loadClass( clazz.getName() + IMPLEMENTATION_SUFFIX ).newInstance();
-            return mapper;
+            Class<T> implementation = (Class<T>) classLoader.loadClass( clazz.getName() + IMPLEMENTATION_SUFFIX );
+            Constructor<T> constructor = implementation.getDeclaredConstructor((Class<?>[]) null);
```

Comment:
Why is `(Class<?>[]) null` passed? Shouldn't be necessary.

Reply:
No. I will remove it